### ISSUE TITLE
Metric LastSuccessfulResponse is updated even if API returns error

### ISSUE THREAD

**Issue by findmyname666** [fastly/fastly-exporter#161]
The Fastly exporter updates the metric [lastSuccessfulResponse](https://github.com/fastly/fastly-exporter/blob/e0f82feb6026ca10de7c2e046ef9632624f0b037/pkg/prom/metrics.go#L26) even if the Fastly API returns an error. In our case, I noticed that the Fastly token was revoked from the log messages:

```
level=error component=rt.fastly.com service_id=_reducted_ status_code=403 response_ts=1712674340 err="invalid authentication" msg="token may be invalid"
```

The metric description suggests that the metric shouldn't be updated:

```
"last_successful_response", Help: "Unix timestamp of the last successful response received from the real-time stats API."
```

I tracked it to a specific branch of the code related to retrieving origin metrics as an example:
- The function [NewMetrics](https://github.com/fastly/fastly-exporter/blob/e0f82feb6026ca10de7c2e046ef9632624f0b037/pkg/prom/metrics.go#L23) (mentioned above) is called [here](https://github.com/fastly/fastly-exporter/blob/e0f82feb6026ca10de7c2e046ef9632624f0b037/pkg/prom/registry.go#L85).
- The function [MetricsFor](https://github.com/fastly/fastly-exporter/blob/main/pkg/prom/registry.go#L78C20-L78C30) is called [here](https://github.com/fastly/fastly-exporter/blob/main/pkg/rt/manager.go#L158) from within the [spawn](https://github.com/fastly/fastly-exporter/blob/main/pkg/rt/manager.go#L156C19-L156C24) function.
- Inside the [spawn](https://github.com/fastly/fastly-exporter/blob/main/pkg/rt/manager.go#L156C19-L156C24) function, the [RunOrigins](https://github.com/fastly/fastly-exporter/blob/main/pkg/rt/manager.go#L164) function is called.
- Inside the [RunOrigins](https://github.com/fastly/fastly-exporter/blob/main/pkg/rt/subscriber.go#L123) function [queryOrigins](https://github.com/fastly/fastly-exporter/blob/main/pkg/rt/subscriber.go#L130) is called, which returns the API result/error as the second return value, but it is ignored. [A few lines down](https://github.com/fastly/fastly-exporter/blob/main/pkg/rt/subscriber.go#L134), the metric `lastSuccessfulResponse` is updated.

Related questions:
- Is this the desired behaviour?
- If yes, should another metric be introduced?

If necessary, I can take a look at the code adjustments once we agree on the next steps.

**Comment by leklund** [comment]
I believe the intent of the `lastSuccessfulResponse` was to indicate that the http request to rt.fastly.com succeeded, regardless of the response status code. If, however, this was the intended behavior I'm not sure I see a use case where you'd want the metric to be set from a 403.

I see two options:
1. add a label to the metric for the status code. This could retain the existing metric but allow clients to filter where status=200.
2. only Set `LastSuccessfulResponse` when the response status code == 200.

**Comment by findmyname666** [comment]
> I believe the intent of the `lastSuccessfulResponse` was to indicate that the http request to rt.fastly.com succeeded, regardless of the response status code. If, however, this was the intended behavior I'm not sure I see a use case where you'd want the metric to be set from a 403.
> 
> I see two options:
> 
> 1. add a label to the metric for the status code. This could retain the existing metric but allow clients to filter where status=200.
> 2. only Set `LastSuccessfulResponse` when the response status code == 200.

I agree with your proposal. Any of those would be helpful. However if we go with the first, there is still the question if the metrics `lastSuccessfulResponse` should report differently for 4xx and 5xx. From my point of view 4xx and 5xx aren't successful responses for a client.
